File: plugin.sync.bug/src/test/scala/com/anarsoft/race/detection/testFixture/TestData.scala
package com.anarsoft.race.detection.testFixture

import com.anarsoft.race.detection.event.gen.VolatileAccessEventGen
import com.anarsoft.race.detection.event.method.LoadedMethodEvent
import com.vmlens.trace.agent.bootstrap.event.RuntimeEvent


case class TestData(syncActionJavaEvents: List[RuntimeEvent], volatileAccessEvents: List[VolatileAccessEventGen],
                    methodJavaEvents: List[RuntimeEvent], methodEvents: List[LoadedMethodEvent]) {

}

File: trace-agent-bootstrap/src/test/java/com/vmlens/trace/agent/bootstrap/testfixture/TestData.java
package com.vmlens.trace.agent.bootstrap.testfixture;

import com.vmlens.trace.agent.bootstrap.interleave.alternatingOrder.AlternatingOrderContainer;
import com.vmlens.trace.agent.bootstrap.interleave.testUtil.FeatureTestMatcher;

public class TestData {

    private ResultTestBuilder resultTestBuilder;
    private AlternatingOrderContainer alternatingOrderContainer;
    private FeatureTestMatcher featureTestMatcher;

    public TestData() {
    }

    public ResultTestBuilder resultTestBuilder() {
        return resultTestBuilder;
    }

    public TestData setResultTestBuilder(ResultTestBuilder resultTestBuilder) {
        this.resultTestBuilder = resultTestBuilder;
        return this;
    }

    public AlternatingOrderContainer alternatingOrderContainer() {
        return alternatingOrderContainer;
    }

    public TestData setAlternatingOrderContainer(AlternatingOrderContainer alternatingOrderContainer) {
        this.alternatingOrderContainer = alternatingOrderContainer;
        return this;
    }

    public FeatureTestMatcher featureTestMatcher() {
        return featureTestMatcher;
    }

    public TestData setFeatureTestMatcher(FeatureTestMatcher featureTestMatcher) {
        this.featureTestMatcher = featureTestMatcher;
        return this;
    }


}
